android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/MainActivity.java

```diff
@@ -1065,17 +1065,22 @@ public void onKeyboardDownloadFinished(HashMap<String, String> keyboardInfo, int
   @Override
   public void onPackageInstalled(List<Map<String, String>> keyboardsInstalled) {
     for(int i=0; i < keyboardsInstalled.size(); i++) {
-      HashMap<String, String> keyboardInfo = new HashMap<>(keyboardsInstalled.get(i));
+      HashMap<String, String> hashMap = new HashMap<>(keyboardsInstalled.get(i));
+      Keyboard keyboardInfo = new Keyboard(
+        hashMap.get(KMManager.KMKey_PackageID),
+        hashMap.get(KMManager.KMKey_KeyboardID),
```

Comment:
There's obviously a little more that could be done here, but we've probably reached an acceptable stopping-point for the current set of changes in this direction, especially noting one of your reply comments elsewhere (re: keyboard downloads being out of current scope).

After all, the longform constructor here is necessitated because the method's parameter is still `Map`-based.  As a result, fixing things here would require propagating even more changes elsewhere, etc.  We can chase that thread another time.